tests/test.validator.js

```diff
@@ -0,0 +1,65 @@
+import Validator from 'validator';
+
+
+
+describe('Validator', function() {
+
+  it('should detect an invalid file with ENOENT', () => {
+    var av = new Validator({_: ['foo']});
+    av.handleError = sinon.stub();
+    var fakeError = new Error('soz');
+    fakeError.code = 'ENOENT';
+    var fakeLstat = () =>  {
+      return Promise.reject(fakeError);
+    };
+    return av.checkFileExists(av.packagePath, fakeLstat)
+      .then(() => {
+        assert.fail(null, null, 'Unexpected success');
+      })
+      .catch(err => {
+        assert.instanceOf(err, Error);
+        assert.include(err.message, 'Path given is not a file');
+      });
+  });
+
+  it('should detect other errors during lstat', () => {
+    var av = new Validator({_: ['foo']});
```

Comment:
Yeah in tests I might do something like this for brevity. Happy to be a bit more verbose though.